lib/Activity.py

```diff
@@ -55,7 +55,9 @@ class ActivityError(Exception):
     'state': None,
     'sessionid': None,
     'status': None,
-    'comment': None
+    'comment': None,
+    'media_dir': None,
```

Comment:
I'd be interested in doing a bigger hammer here, like storing all the output of the following.  This will get us media_dir, plus lots of other stuff we may wish to have access to:
```
ncn-m001:~ # kubectl get configmaps -n argo lemondrop-alice-iw1 -o jsonpath='{.data.iuf_activity}' | jq '{"name", "input_parameters", "site_parameters", "operation_outputs", "activity_state"}'
{
  "name": "lemondrop-alice-iw1",
  "input_parameters": {
    "media_dir": "/lemondrop-alice-iw1",
    "site_parameters": "",
    "limit_management_nodes": [
      "Management_Worker"
    ],
    "limit_managed_nodes": [
      "Compute"
    ],
    "managed_rollout_strategy": "stage",
    "concurrent_management_rollout_percentage": 20,
    "media_host": "ncn-m001",
    "concurrency": 0,
    "bootprep_config_managed": ".bootprep-lemondrop-alice-iw1/compute-and-uan-bootprep.yaml",
    "bootprep_config_management": ".bootprep-lemondrop-alice-iw1/management-bootprep.yaml",
    "stages": [
      "process-media"
    ],
    "force": false
  },
  "site_parameters": {
    "global": {},
    "products": {
      "HFP-firmware": {
        "internal_dac": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-master-local/dev/master/noos/noarch/HFP-DOCS/HFP-DOCS-pdfhtml-release-HFP-22.10.0-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/HFP-firmware/hfp-Alice-23/HFP-firmware-23.1.2.tar.gz",
        "version": "23.1.2"
      },
      "analytics": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-stable-local/release/analytics-1.4/noos/noarch/Analytics/Analytics-pdfhtml-release-Analytics-1.4-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/analytics/analytics-1.4.18.tar.gz",
        "version": "1.4.18",
        "working_branch": "integration-1.4.18"
      },
      "cos": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-stable-local/release/cos-2.5/noos/noarch/COS/COS-pdfhtml-release-cos-2.5-LATEST.tar",
        "internal_hpcm": "https://arti.dev.cray.com/artifactory/mss-distribution-stable-local/cos/cos-2.5/cos-2.5.87-sles15sp4-x86_64.iso",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/cos/cos-2.5/cos-2.5.91.tar.gz",
        "version": "2.5.95",
        "working_branch": "integration-2.5.95"
      },
      "cpe": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-master-local/dev/master/noos/noarch/CPE/CPE-pdfhtml-23.02-20230209153233_8e10109.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/cpe/23.02/cpe-23.02-sles15-sp4-csm-23.2.2.tar.gz",
        "version": "23.2.2",
        "working_branch": "integration-23.2.2"
      },
      "cray-sdu-rda": {
        "internal_dac": "https://cflmetint01.hpc.amslabs.hpecorp.net/public-html/hpc-ch-sdu-tarfile/shasta/2.2.0-beta15/",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net:443/artifactory/shasta-distribution-unstable-local/sdu/cray-sdu-rda-2.2.0-beta15.tar.gz",
        "version": "2.2.0-beta15"
      },
      "csm": {
        "version": "1.4.0-beta.40"
      },
      "csm-diags": {
        "internal_dac": "https://github.hpe.com/hpe/hpc-csmdiags/tree/master/dac-assets/documents",
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-master-local/dev/master/noos/noarch/csmdiags-docs/csmdiags-docs-pdfhtml-release-csmdiags-1.4-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/csm-diags/csm-diags-1.4.0/csm-diags-1.4.9.tar.gz",
        "version": "1.4.9"
      },
      "default": {
        "network_type": "mellanox",
        "note": "",
        "suffix": "-iw1b-pbs-slurm",
        "wlm": "future use",
        "working_branch": "integration-{{version_x_y_z}}"
      },
      "hfp": {
        "version": ""
      },
      "pbs": {
        "internal_dac": "https://rndwiki-pro.its.hpecorp.net/display/CASMUSER/WLM+IUF+Installation",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-unstable-local/wlm-pbs/cpe-pbs-23.02-sles15-1.2.9-20230209172608.28ca7d3.tar.gz",
        "version": "1.2.9-20230209172608.28ca7d3",
        "working_branch": "integration-1.2.9"
      },
      "recipe": {
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-unstable-local/hpc-csm-software-recipe/hpc-csm-software-recipe-22.11.0-alpha.15.20230202172927-fe48a62.tar.gz",
        "version": "23.2.16"
      },
      "sat": {
        "internal_docs": "https://cray-hpe.github.io/docs-sat/en-25/",
        "internal_media": "https://release.algol60.net/sat/sat/sat-2.5.13.tar.gz",
        "version": "2.5.13"
      },
      "sle-os-backports-sle-15-sp4-x86_64": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-stable-local/release/15-SP4_cos2.5/noos/noarch/SLE/SLE-pdfhtml-release-SLE-23.2.7-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/os/15-SP4/COS2.5/x86_64/23.2.7/SUSE-Backports-SLE-15-SP4-x86_64-23.2.7-20230106.tar.gz",
        "version": "23.2.7-20230106"
      },
      "sle-os-products-15-sp4-x86_64": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-stable-local/release/15-SP4_cos2.5/noos/noarch/SLE/SLE-pdfhtml-release-SLE-23.2.7-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/os/15-SP4/COS2.5/x86_64/23.2.7/SUSE-Products-15-SP4-x86_64-23.2.7-20230106.tar.gz",
        "version": "23.2.7-20230106"
      },
      "sle-os-updates-15-sp4-x86_64": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/pubs-misc-stable-local/release/15-SP4_cos2.5/noos/noarch/SLE/SLE-pdfhtml-release-SLE-23.2.7-LATEST.tar",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/os/15-SP4/COS2.5/x86_64/23.2.7/SUSE-Updates-15-SP4-x86_64-23.2.7-20230106.tar.gz",
        "version": "23.2.7-20230106"
      },
      "sleupdates-15-SP4-23.2.7-20230106-supplement": {
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/os/15-SP4/COS2.5/x86_64/23.2.7/sleupdates-15-SP4-23.2.7-20230106-supplement.tar.gz",
        "version": "23.2.7-20230106"
      },
      "slingshot": {
        "internal_docs": "https://arti.hpc.amslabs.hpecorp.net/artifactory/slingshot-misc-stable-local/external/release/2.0.1/RC4/slingshot-docs-2.0.1.tar.gz",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/slingshot-misc-stable-local/external/release/2.0.2/PRE-RC2/slingshot-Cray-EX-2.0.2-803.tar.gz",
        "version": "2.0.2-803"
      },
      "slingshot-host-software": {
        "internal_media1": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-unstable-local/slingshot-host-software/slingshot-host-software-2.0.2-91-cos-2.5.tar.gz",
        "internal_media2": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-unstable-local/slingshot-host-software/slingshot-host-software-2.0.2-91-csm-1.4.tar.gz",
        "version": "2.0.2-91-csm-1.4",
        "working_branch": "integration-2.0.2"
      },
      "slurm": {
        "internal_dac": "https://rndwiki-pro.its.hpecorp.net/display/CASMUSER/WLM+IUF+Installation",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-unstable-local/wlm-slurm/cpe-slurm-23.02-sles15-1.2.9-20230209172621.c445c79.tar.gz",
        "version": "1.2.9-20230209172621.c445c79",
        "working_branch": "integration-1.2.9"
      },
      "sma": {
        "internal_dac": "https://github.hpe.com/hpe/hpc-car-sma-docs/blob/CASMSMF-7123-IUF/dac-assets/documents/install/Install_the_System_Monitoring_Application_Product_Stream.md",
        "internal_media": "https://arti.hpc.amslabs.hpecorp.net/artifactory/shasta-distribution-stable-local/sma/sma-1.8/sma-1.8.4.tar.gz",
        "version": "1.8.6",
        "working_branch": "integration-1.8.6"
      },
      "uan": {
        "internal_docs": "https://github.com/Cray-HPE/docs-uan",
        "internal_media": "https://release.algol60.net/uan/uan-2.6.0-alpha.5.tar.gz",
        "version": "2.6.0-alpha.5",
        "working_branch": "integration-2.6.0"
      },
      "wlm-slurm": {
        "version": "1.2.9"
      }
    }
  },
  "operation_outputs": {
    "stage_params": {
      "managed-nodes-rollout": {
        "managed-nodes-rollout": {
          "build-sat-general-subcommand": {
            "sat_command": "sat bootsys --bos-version v2 reboot --disruptive --stage bos-operations --bos-templates compute-23.2.16-iw1b-pbs-slurm,uan-23.2.16-iw1b-pbs-slurm,lnet-23.2.16-iw1b-pbs-slurm --bos-shutdown-timeout 3000 --bos-boot-timeout 3000 --bos-limit x3000c0s21b0n0"
          },
          "prom-metrics": {
            "diff-time-value": "1489"
          },
          "sat-bootsys-reboot": {
            "script_stdout": ""
          }
        }
      },
      "management-nodes-rollout": {
        "management-nodes-rollout": {
          "prom-metrics": {
            "diff-time-value": "63"
          },
          "rebuild-m002(0)": {
            "output": ""
          },
          "rebuild-m003(0)": {
            "output": ""
          }
        }
      },
      "pre-install-check": {
        "preflight-checks-for-services": {
          "preflight-checks(0)": {
            "output": ""
          },
          "prom-metrics": {
            "diff-time-value": "33"
          }
        }
      },
      "prepare-images": {
        "prepare-managed-images": {
          "prom-metrics": {
            "diff-time-value": "3861"
          },
          "sat-bootprep-run": {
            "script_stdout": "{\n    \"images\": [\n        {\n            \"name\": \"compute-cray-shasta-compute-sles15sp4.noarch-2.5.30-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"e4dbe74a-1271-42fa-8dbe-24f7759c70be\",\n            \"configuration\": \"compute-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Compute\"\n            ]\n        },\n        {\n            \"name\": \"gpu-compute-cray-shasta-compute-sles15sp4.noarch-2.5.30-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"450174d8-fe6e-4732-a1c9-a31eda78ad06\",\n            \"configuration\": \"gpu-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Compute\"\n            ]\n        },\n        {\n            \"name\": \"uan-cray-shasta-compute-sles15sp4.noarch-2.5.30-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"e3c59cdd-cb3b-4f00-aaa1-fe6d45872cf3\",\n            \"configuration\": \"uan-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Application\",\n                \"Application_UAN\"\n            ]\n        },\n        {\n            \"name\": \"lnet-cray-shasta-compute-sles15sp4.noarch-2.5.30-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"dae90339-4492-4d4f-ba17-12df1fc18552\",\n            \"configuration\": \"lnet-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Application\",\n                \"Application_LNETRouter\"\n            ]\n        },\n        {\n            \"name\": \"cray-shasta-compute-sles15sp4.noarch-2.5.30-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": \"ac5938c8-921a-47af-b6e4-dd5bfcc91a86\",\n            \"final_image_id\": \"ac5938c8-921a-47af-b6e4-dd5bfcc91a86\",\n            \"configuration\": null,\n            \"configuration_group_names\": null\n        }\n    ],\n    \"session_templates\": [\n        {\n            \"name\": \"compute-23.2.16-iw1b-pbs-slurm\",\n            \"configuration\": \"compute-23.2.16-iw1b-pbs-slurm\"\n        },\n        {\n            \"name\": \"uan-23.2.16-iw1b-pbs-slurm\",\n            \"configuration\": \"uan-23.2.16-iw1b-pbs-slurm\"\n        },\n        {\n            \"name\": \"lnet-23.2.16-iw1b-pbs-slurm\",\n            \"configuration\": \"lnet-23.2.16-iw1b-pbs-slurm\"\n        }\n    ]\n}"
          }
        },
        "prepare-management-images": {
          "prom-metrics": {
            "diff-time-value": "732"
          },
          "sat-bootprep-run": {
            "script_stdout": "{\n    \"images\": [\n        {\n            \"name\": \"worker-secure-kubernetes-0.4.41-x86_64.squashfs-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"8e3b5f88-b54b-4ac1-a4a6-cd59d7376ff8\",\n            \"configuration\": \"management-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Management_Worker\"\n            ]\n        },\n        {\n            \"name\": \"master-secure-kubernetes-0.4.41-x86_64.squashfs-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"627c095f-375f-4aef-b1be-b81ea09d6504\",\n            \"configuration\": \"management-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Management_Master\"\n            ]\n        },\n        {\n            \"name\": \"storage-secure-storage-ceph-0.4.41-x86_64.squashfs-iw1b-pbs-slurm\",\n            \"preconfigured_image_id\": null,\n            \"final_image_id\": \"b20db953-0cd1-4fc9-9e13-e5302efc6c31\",\n            \"configuration\": \"management-23.2.16-iw1b-pbs-slurm\",\n            \"configuration_group_names\": [\n                \"Management_Storage\"\n            ]\n        }\n    ]\n}"
          }
        }
      },
      "process-media": {
        "products": {
          "HFP-firmware-23-1-2": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/HFP-firmware-23.1.2"
          },
          "analytics-1-4-18": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/analytics-1.4.18"
          },
          "cos-2-5-91": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/cos-2.5.91"
          },
          "cos-2-5-95": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/cos-2.5.95"
          },
          "cpe-23-2-2": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/cpe-23.02-sles15-sp4"
          },
          "cray-sdu-rda-2-2-0-beta15": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/cray-sdu-rda-2.2.0-beta15"
          },
          "csm-diags-1-4-9": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/csm-diags-1.4.9"
          },
          "sat-2-5-13": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/sat-2.5.13"
          },
          "sle-os-backports-sle-15-sp4-x86_64-23-2-7-20230106": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/SUSE-Backports-SLE-15-SP4-x86_64-23.2.7-20230106"
          },
          "sle-os-products-15-sp4-x86_64-23-2-7-20230106": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/SUSE-Products-15-SP4-x86_64-23.2.7-20230106"
          },
          "sle-os-updates-15-sp4-x86_64-23-2-7-20230106": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/SUSE-Updates-15-SP4-x86_64-23.2.7-20230106"
          },
          "slingshot-2-0-2-803": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/slingshot-2.0.2-803"
          },
          "slingshot-host-software-2-0-2-91-cos-2-5": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/slingshot-host-software-2.0.2-91-cos-2.5"
          },
          "slingshot-host-software-2-0-2-91-csm-1-4": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/slingshot-host-software-2.0.2-91-csm-1.4"
          },
          "sma-1-8-6": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/sma-1.8.6"
          },
          "uan-2-6-0-alpha-5": {
            "parent_directory": "/etc/cray/upgrade/csm/lemondrop-alice-iw1/uan-2.6.0-alpha.5"
          }
        }
      },
      "update-cfs-config": {
        "update-managed-cfs-config": {
          "prom-metrics": {
            "diff-time-value": "39"
          },
          "sat-bootprep-run": {
            "script_stdout": "{\n    \"configurations\": [\n        {\n            \"name\": \"compute-23.2.16-iw1b-pbs-slurm\"\n        },\n        {\n            \"name\": \"lnet-23.2.16-iw1b-pbs-slurm\"\n        },\n        {\n            \"name\": \"uan-23.2.16-iw1b-pbs-slurm\"\n        },\n        {\n            \"name\": \"gpu-23.2.16-iw1b-pbs-slurm\"\n        }\n    ]\n}"
          }
        },
        "update-management-cfs-config": {
          "prom-metrics": {
            "diff-time-value": "46"
          },
          "sat-bootprep-run": {
            "script_stdout": "{\n    \"configurations\": [\n        {\n            \"name\": \"management-23.2.16-iw1b-pbs-slurm\"\n        }\n    ]\n}"
          }
        }
      }
    }
  },
  "activity_state": "wait_for_admin"
}
ncn-m001:~ #
```